lnwire/features.go

```diff
@@ -35,6 +35,11 @@ const (
 	// connection is established.
 	InitialRoutingSync FeatureBit = 3
 
+	// UpfrontShutdownScript indicates that the node supports committing upfront
+	// to the shutdown script when opening channels.  This prevents a node that
+	// has been compromised from having all the channel funds stolen.
+	UpfrontShutdownScript FeatureBit = 5
```

Comment:
we should have required and a optional bit, similar to the other fields.

Reply:
indeed, they need to be introduced in pairs otherwise we can't tell if the sender is _requiring_ this option. we made a similar mistake in adding the gossip queries back in the day

Reply:
I'll add the even bit as well. 

In terms of mandatory bits, if a peer has the mandatory upfront shutdown bit set it just means that we have to support that feature for them, right? Not that we **have to** set a shutdown addr ourselves.

Reply:
yep, I think so. If they require it, they want to know that we can support honouring their wish.

Reply:
Ok cool, that makes sense :)